lib/request-processor.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+var _ = require('underscore')
+  , parser = require('./parser')
+  , Subscriber = require('./subscriber')
+  , messageCreator = require('./message-creator');
+
+var subscriptionHandlers = {
+  subscribe: function (phoneNumber, movie) {
+    Subscriber.find(phoneNumber)
+      .then(function(subscriber) {
+        var movies = addMovie(subscriber, movie);
+        Subscriber.create(phoneNumber, movies);
+      });
+  },
+
+  unsub: function (phoneNumber, movie) {
+    Subscriber.find(phoneNumber)
+      .then(function(subscriber) {
+        if (subscriber) {
```

Comment:
This is an interesting question and a tough one as well.

Let me provide some context, this object represents a _subscriber_.

```javascript
{
  id: 'SXXXXXXXXX',
  movies: ['han_solo_spinoff', 'rogue_one']
}
```

If we want to remove `'rogue_one'` this handler will take care of this, after removing that subscription the **new** subscriptor will be:

```javascript
{
  id: 'SXXXXXXXXX',
  movies: ['han_solo_spinoff']
}
```

Then, we want to remove `'han_solo_spinoff'`, the **new** subscriptor will be:

```javascript
{
  id: 'SXXXXXXXXX',
  movies: []
}
```

*But*, there is no sense in having a subscriber without subscriptions, we use `Subscriber.delete` to take care of that work.

Finally _and the most important part_,

If there is any subscriber, there is no valid operation we can use it. What should be the best approach to make this function total? It's worthy to make it that way?